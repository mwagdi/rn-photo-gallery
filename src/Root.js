import React, { Component } from 'react';
import {
    StyleSheet,
    Text,
    View,
    ScrollView,
    Modal,
    TouchableHighlight,
    Image,
    TextInput,
    Button
} from 'react-native';
import { Icon } from 'react-native-elements';
import PhotoListContainer from './containers/PhotoListContainer';
import { connect } from 'react-redux';
import { closeModal, addComment } from './store/actions'

class Root extends Component{
    state = {
        commentInput: ""
    }
    render(){
        const { photo, open, closeModal, addComment } = this.props;
        return (
            <View style={styles.container}>
                <Text style={styles.title}>Photo Gallery</Text>
                <ScrollView style={styles.scrollview}>
                    <PhotoListContainer />
                </ScrollView>
                <Modal
                animationType="slide"
                transparent={false}
                visible={open}
               >
                <View style={{marginTop: 22, padding: 10}}>
                    <View>
                    <Text>{photo.title}</Text>

                    <TouchableHighlight
                        style={{position: 'absolute', right: 0}}
                        onPress={() => {
                            closeModal();
                            this.setState({ commentInput: "" })
                        }}>
                        <Icon
                            name='close' />
                    </TouchableHighlight>
                    <Image style={styles.image} source={{uri: photo.src}} />
                    {photo.comments &&
                    <View>
                        <Text>Comments</Text>
                        {photo.comments.map((comment, i) => <Text key={`comment${i}`}>{comment}</Text>)}
                        <TextInput
                            style={{
                                height: 40,
                                borderColor: 'gray',
                                borderWidth: 1,
                                marginBottom: 10
                            }}
                            onChangeText={commentInput => this.setState({commentInput})}
                            value={this.state.commentInput}
                        />
                        <Button
                        onPress={() => {
                            addComment(photo.id, this.state.commentInput);
                            this.setState({ commentInput: "" })
                        }}
                        title="ADD COMMENT"
                        color="#1f67db"
                        />
                    </View>}
                    
                    </View>
                </View>
                </Modal>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: 40,
      backgroundColor: '#fff',
      alignItems: 'center',
      justifyContent: 'center',
    },
    scrollview: {
        flex: 1
    },
    title: {
        fontSize: 15,
        fontWeight: "bold",
        width: "100%",
        textAlign: "center"
    },
    image: {
        width: "100%",
        height: 300
    }
});

const mapStateToProps = state => {
    const {  open, photo } = state.modal
    return {
        open,
        photo
    }
}

export default connect(mapStateToProps, { closeModal, addComment })(Root);