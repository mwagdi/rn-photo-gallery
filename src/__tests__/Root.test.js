import React from 'react';
import renderer from 'react-test-renderer';
import Root from '../Root';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('<Root />', () => {
  const initialState = {
    modal: {
      open: false
    }
  }; 
  const mockStore = configureStore();
  let store = mockStore(initialState);
  let wrapper = shallow(<Root store={store} />);
  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});