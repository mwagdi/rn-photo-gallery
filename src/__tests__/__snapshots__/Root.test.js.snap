// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> renders correctly 1`] = `
<ContextProvider
  value={null}
>
  <Root
    addComment={[Function]}
    closeModal={[Function]}
    open={false}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
